### ISSUE TITLE
git push couses everything to freeze

### ISSUE THREAD

**Issue by laycookie** [hyprwm/Hyprland#6783]
### Regression?

Yes

### System Info and Version

<details>

```sh
Hyprland, built from branch main at commit 918d8340afd652b011b937d29d5eea0be08467f5  (flake.lock: update).
Date: 2024-06-25
Tag: v0.41.2, commits: 918d8340afd652b011b937d29d5eea0be08467f5

flags: (if any)


System Information:
System name: Linux
Node name: nixos
Release: 6.6.36
Version: #1-NixOS SMP PREEMPT_DYNAMIC Thu Jun 27 11:49:15 UTC 2024


GPU information: 
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation GP104 [GeForce GTX 1080] [10de:1b80] (rev a1) (prog-if 00 [VGA controller])


os-release: ANSI_COLOR="1;34"
BUG_REPORT_URL="https://github.com/NixOS/nixpkgs/issues"
BUILD_ID="24.11.20240703.9f4128e"
DOCUMENTATION_URL="https://nixos.org/learn.html"
HOME_URL="https://nixos.org/"
ID=nixos
IMAGE_ID=""
IMAGE_VERSION=""
LOGO="nix-snowflake"
NAME=NixOS
PRETTY_NAME="NixOS 24.11 (Vicuna)"
SUPPORT_URL="https://nixos.org/community.html"
VERSION="24.11 (Vicuna)"
VERSION_CODENAME=vicuna
VERSION_ID="24.11"


plugins:

======Config-Start======


Config File: /home/Dennis/.config/hypr/hyprland.conf: Read Succeeded
exec-once = /nix/store/y29cbwfiwb9nyi4db4apk9dgylzdjm1v-dbus-1.14.10/bin/dbus-update-activation-environment --systemd DISPLAY HYPRLAND_INSTANCE_SIGNATURE WAYLAND_DISPLAY XDG_CURRENT_DESKTOP && systemctl --user stop hyprland-session.target && systemctl --user start hyprland-session.target
decoration {
  col.shadow=rgba(28282899)
}

general {
  col.active_border=rgb(83a598)
  col.inactive_border=rgb(665c54)
}

group {
  groupbar {
    col.active=rgb(83a598)
    col.inactive=rgb(665c54)
    text_color=rgb(d5c4a1)
  }
  col.border_active=rgb(83a598)
  col.border_inactive=rgb(665c54)
  col.border_locked_active=rgb(8ec07c)
}

misc {
  background_color=rgb(282828)
}
# Monitor
monitor=DP-2,3840x2160@60,auto,1.5
monitor=DP-3,2560x1440@144,auto,1 

xwayland {
    force_zero_scaling = true
}

exec = pkill waybar & sleep 0.5 && waybar
exec-once = swww init & sleep 0.5 && exec wallpaper_random
# exec-once = wallpaper_random

# Set en layout at startup

# Input config
input {
    kb_layout = us,ru
    kb_variant =
    kb_model =
    kb_options = grp:alt_shift_toggle
    kb_rules =

    follow_mouse = 1

    touchpad {
        natural_scroll = false
    }

    sensitivity = 0 # -1.0 - 1.0, 0 means no modification.
}

general {

    gaps_in = 5
    gaps_out = 20
    border_size = 2
    col.active_border = rgb(fbf1c7)
    col.inactive_border = rgba(595959aa)

    layout = dwindle
}

decoration {
    rounding = 10

    drop_shadow = true
    shadow_range = 4
    shadow_render_power = 3
    col.shadow = rgba(1a1a1aee)
}

animations {
    enabled = yes

    bezier = ease,0.4,0.02,0.21,1

    animation = windows, 1, 3.5, ease, slide
    animation = windowsOut, 1, 3.5, ease, slide
    animation = border, 1, 6, default
    animation = fade, 1, 3, ease
    animation = workspaces, 1, 3.5, ease
}

dwindle {
    pseudotile = yes
    preserve_split = yes
}

# master {
#     new_is_master = yes
# }

gestures {
    workspace_swipe = false
}

# Example windowrule v1
# Example windowrule v2

windowrule=float,^(pavucontrol)$
windowrule=float,^(blueman-manager)$
windowrule=size 934 525,^(mpv)$
windowrule=float,^(mpv)$
windowrule=center,^(mpv)$
#windowrule=pin,^(firefox)$

$mainMod = SUPER
bind = $mainMod, G, fullscreen,


bind = $mainMod, SPACE, 	exec,	rofi -show drun
bind = $mainMod, M, 	exec,	wlogout
bind = $mainMod, Q, 	exec,	kitty
bind = $mainMod, D, 	exec,	webcord
bind = $mainMod, S, 	exec,	steam
bind = $mainMod, E, 	exec,	dolphin
bind = $mainMod, T, 	exec,	telegram-desktop
bind = $mainMod, B, 	exec,	firefox	
bind = $mainMod, C,		killactive,
bind = $mainMod, F, 	togglefloating,
bind = $mainMod, V, 	togglesplit,

# Move focus
bind = $mainMod, H, movefocus, l
bind = $mainMod, L, movefocus, r
bind = $mainMod, K, movefocus, u
bind = $mainMod, J, movefocus, d

bind = $mainMod SHIFT, H, movewindow, l
bind = $mainMod SHIFT, L, movewindow, r
bind = $mainMod SHIFT, K, movewindow, u
bind = $mainMod SHIFT, J, movewindow, d

bind = , Print, exec, grim -g "$(slurp)" - | wl-copy
bind = SHIFT, Print, exec, grim -g "$(slurp)"

# Functional keybinds
bind =,XF86AudioMicMute,exec,pamixer --default-source -t
bind =,XF86MonBrightnessDown,exec,light -U 20
bind =,XF86MonBrightnessUp,exec,light -A 20
bind =,XF86AudioMute,exec,pamixer -t
bind =,XF86AudioLowerVolume,exec,pamixer -d 10
bind =,XF86AudioRaiseVolume,exec,pamixer -i 10
bind =,XF86AudioPlay,exec,playerctl play-pause
bind =,XF86AudioPause,exec,playerctl play-pause

# to switch between windows in a floating workspace
bind = SUPER,Tab,cyclenext,
bind = SUPER,Tab,bringactivetotop,

# Move focus with mainMod + arrow keys
bind = $mainMod, left, movefocus, l
bind = $mainMod, right, movefocus, r
bind = $mainMod, up, movefocus, u
bind = $mainMod, down, movefocus, d

# Switch workspaces with mainMod + [0-9]
bind = $mainMod, 1, workspace, 1
bind = $mainMod, 2, workspace, 2
bind = $mainMod, 3, workspace, 3
bind = $mainMod, 4, workspace, 4
bind = $mainMod, 5, workspace, 5
bind = $mainMod, 6, workspace, 6
bind = $mainMod, 7, workspace, 7
bind = $mainMod, 8, workspace, 8
bind = $mainMod, 9, workspace, 9
bind = $mainMod, 0, workspace, 10

# Move active window to a workspace with mainMod + SHIFT + [0-9]
bind = $mainMod SHIFT, 1, movetoworkspace, 1
bind = $mainMod SHIFT, 2, movetoworkspace, 2
bind = $mainMod SHIFT, 3, movetoworkspace, 3
bind = $mainMod SHIFT, 4, movetoworkspace, 4
bind = $mainMod SHIFT, 5, movetoworkspace, 5
bind = $mainMod SHIFT, 6, movetoworkspace, 6
bind = $mainMod SHIFT, 7, movetoworkspace, 7
bind = $mainMod SHIFT, 8, movetoworkspace, 8
bind = $mainMod SHIFT, 9, movetoworkspace, 9
bind = $mainMod SHIFT, 0, movetoworkspace, 10

# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1

# Move/resize windows with mainMod + LMB/RMB and dragging
bindm = $mainMod, mouse:272, movewindow
bindm = $mainMod, mouse:273, resizewindow
bindm = ALT, mouse:272, resizewindow

======Config-End========
```
</details>

### Description

When I try to run `git push` to push my code, everything just fezzes. Not even sure how to figure out what is going on. Im also unable to access the console using `ctrl+alt+3`. 

### How to reproduce

Just `git push`

### Crash reports, logs, images, videos

No crash reports were generated
[918d8340afd652b011b937d29d5eea0be08467f5_1720240707_1264723904.zip](https://github.com/user-attachments/files/16114901/918d8340afd652b011b937d29d5eea0be08467f5_1720240707_1264723904.zip)

**Comment by vaxerski** [comment]
is this even a hyprland issue?

**Comment by laycookie** [comment]
> is this even a hyprland issue?

I'm not sure but I don't see what else it can be, if there are any other suspects throw them at me. I had difficulty downgrading just hyprland yesterday, but I will try again later today. (It does go away if I use a previous generation on nixOS)

**Comment by laycookie** [comment]
Can confirm when downgrading to `e93fbd7c4f991cb8ef03e433ccc4d43587923e15` the issue goes away. (Working on figuring out what commit causes this behavior)

**Comment by vaxerski** [comment]
since when does git push spawn a window?

Also some hangs have been fixed on git, try that

**Comment by gulafaran** [comment]
> since when does git push spawn a window?
> 
> Also some hangs have been fixed on git, try that

Might be ssh x11 askpass and the cyclic loop thing, should work on -git

**Comment by izmyname** [comment]
Was it fixed for ya?

**Comment by vaxerski** [comment]
closing as fixed on git

**Comment by gulafaran** [comment]
> Which commit exactly contains the fix? I have the same issue on 0.41.2, everytime x11-ssh-askpass is spawned by git, hyprland enters a busy loop. I'd like to apply the patch locally until there's a new release.

https://github.com/hyprwm/Hyprland/commit/d7ea1b7785f31e9fd6477490309721a19a835592
